### PR TITLE
Upload and view trainee resume

### PR DESCRIPTION
**Pull request by aimeudoxie**
## PR Description

This feature allows users to upload or view their resumes within the application. Users have the option to either upload a PDF file directly or provide a link to an externally hosted resume. The main objective is to offer users flexibility in how they manage and present their resumes.

### Upload Resume as PDF:
   - Users can upload their resume as a PDF file directly to their profile
   - The application validates the file type to ensure only PDFs are accepted.
   - Uploaded files are securely stored, and users can later view or replace them.
   
### Provide Resume Link:
   - As an alternative to file uploads, users can provide a URL linking to their resume hosted on an external platform (e.g., Google Drive, LinkedIn).

### View Uploaded Resume:
   - Users can view their uploaded resume (PDF or link) from their profile page.

### Upload or update Resume:
   - Users have the ability to update or replace their resume at any time, either by uploading a new PDF file or providing a new link.

## How has this been tested?
1. Navigate to the " Edit Profile"  section.
2. Choose 'Upload Resume' tab
3. Choose to either upload a PDF file or provide a link.
4. Go back to user profile to view uploaded resume

##Screenshots

WhatsApp Image 2024-10-29 at 00 06 20_084db36f
WhatsApp Image 2024-10-29 at 00 06 21_56fe957f
WhatsApp Image 2024-10-29 at 00 06 21_764735b1

This feature provides a user-friendly way for users to manage their resumes within the platform, enhancing profile customization and professional representation.

### CODE REVIEW

**Review by MugemaneBertin2001 — CHANGES_REQUESTED** [review]
Update cache upon successfully uploading `RESUME` to reflect on profile Directly

**Comment by aimeudoxie** [comment]
> Update cache upon successfully uploading `RESUME` to reflect on profile Directly

Done

**Review by MugemaneBertin2001 — APPROVED** [review]

**Review by niyibi250 — APPROVED** [review]

**Review by MugemaneBertin2001 — CHANGES_REQUESTED** [review]
@aimeudoxie  Please remove all alert replace them with Toasts
SweatsGIF

**Comment by aimeudoxie** [comment]
> @aimeudoxie Please remove all alert replace them with Toasts 

Done

**Review by MugemaneBertin2001 — APPROVED** [review]

**Review by Christian-Ishimwe — APPROVED** [review]

File: app/dashboard/trainee/Profile.tsx [hunk 1]
```diff
@@ -73,6 +73,47 @@ export default function Profile() {
...
+  },
+  skip: !userToken,
+  variables: {
+    orgToken: orgToken,
+  },
+});
+
+useEffect(() => {
+  if (err) {
+    console.log('Error:', err);
```
Comment (ncutixavier) [inline app/dashboard/trainee/Profile.tsx]: remove all logs

**Review by ncutixavier — COMMENTED** [review]
Remove all logs

File: app/dashboard/trainee/profile/index.tsx [hunk 2]
```diff
@@ -66,12 +67,57 @@ export default function Profile() {
...
-        <CoverImage cover={profile?.cover} name={profile?.name} />
-        <TouchableOpacity className="absolute bottom-1 right-4 rounded-full shadow-md">
+        <Image
+          source={
+            profile.cover ? { uri: profile.cover } : require('@/assets/images/background.png')
+          }
+          className="w-full h-full object-cover"
+        />
+        <TouchableOpacity onPress={() => router.push('/dashboard/trainee/profile/edit')} className="absolute bottom-1 right-4 rounded-full shadow-md">
           <SvgXml xml={editBG} />
```
Comment (fredshema) [inline app/dashboard/trainee/profile/index.tsx]: Revert these changes, both the cover image and edit profile routes were updated

File: graphql/queries/user.ts [hunk 3]
```diff
@@ -48,3 +48,21 @@ export const GET_PROFILE = gql`
     }
   }
 `;
+export const GET_TRAINEE_PROFILE = gql`
```
Comment (fredshema) [inline graphql/queries/user.ts:51]: Instead of creating another query you can add resume in the `GET_PROFILE` query and use the only one query to fetch profile

**Review by fredshema — CHANGES_REQUESTED** [review]

File: app/dashboard/trainee/profile/index.tsx (hunk 2 above)
Comment (aimeudoxie) [reply app/dashboard/trainee/profile/index.tsx]: Resolved

**Review by fredshema — APPROVED** [review]

**Review by ncutixavier — APPROVED** [review]
